### PR TITLE
Support ALTER TABLE t REORGANIZE PARTITION

### LINKED ISSUE #8113: Support ALTER TABLE REORGANIZE PARTITION
### Is your feature request related to a problem?

Reorganize partition is going to be supported in TiDB, https://github.com/pingcap/tidb/issues/38535 and TiCDC needs to support it as well.

### Describe the feature you'd like

Currently timodel.ActionReorganizePartition is skipped and needs to be handled, just like other partitioning operations.

### Describe alternatives you've considered

not supporting it?

### Teachability, Documentation, Adoption, Migration Strategy

Nothing more than enable it is needed, since the existing implementation of partition management already supports this.

### PR DESCRIPTION
**Pull request by mjonss**
### What problem does this PR solve?

Issue Number: close #8113

### What is changed and how it works?
Adding `timodel.ActionReorganizePartition` to the `allowDDLList` and include it for `updatePartition` in `DoHandleDDL`.

#### Tests 

 - Unit test
 - Integration test

### Release note 

```release-note
Adding support for ALTER TABLE ... REORGANIZE PARTITION
```

### CODE REVIEW

**Comment by asddongmen** [comment]
Could you please fix the failed unit tests and add a `ALTER TABLE ... REORGANIZE PARTITION` SQL to here: https://github.com/pingcap/tiflow/blob/master/tests/integration_tests/partition_table/data/prepare.sql#L28 to test whether TiCDC can handle this DDL expectedly.

**Comment by mjonss** [comment]
> Could you please fix the failed unit tests and add a `ALTER TABLE ... REORGANIZE PARTITION` SQL to here: https://github.com/pingcap/tiflow/blob/master/tests/integration_tests/partition_table/data/prepare.sql#L28 to test whether TiCDC can handle this DDL expectedly.

Hi @asddongmen, I have added such test as well. Could you please review it now?

**Comment by mjonss** [comment]
@asddongmen @zhaoxinyu @tangenta PTAL, I have updated the patch and included the requested test.

File: tests/integration_tests/partition_table/data/prepare.sql [hunk 1]
```diff
@@ -26,4 +26,7 @@ ALTER TABLE t1 EXCHANGE PARTITION p3 WITH TABLE t2;
 insert into t2 values (100),(101),(102),(103),(104),(105); /*these values will be replicated to in downstream t2*/
 insert into t1 values (25),(29); /*these values will be replicated to in downstream t1.p3*/
 
+ALTER TABLE t1 REORGANIZE PARTITION p0,p2 INTO (PARTITION p0 VALUES LESS THAN (5), PARTITION p1 VALUES LESS THAN (10), PARTITION p2 VALUES LESS THAN (21));
+ALTER TABLE t1 REORGANIZE PARTITION p2,p3,p4 INTO (PARTITION p2 VALUES LESS THAN (20), PARTITION p3 VALUES LESS THAN (26), PARTITION p4 VALUES LESS THAN (35), PARTITION pMax VALUES LESS THAN (MAXVALUE));
+
```
Comment (asddongmen) [inline tests/integration_tests/partition_table/data/prepare.sql:31]: Please add some insert querys below line 29 and line 30 to examine TiCDC can replicate data correctly after these DDL. thanks!

**Review by zhaoxinyu — APPROVED** [review]

File: tests/integration_tests/partition_table/data/prepare.sql (hunk 1 above)
Comment (asddongmen) [reply tests/integration_tests/partition_table/data/prepare.sql:37]: thank you ~! LGTM

**Review by asddongmen — APPROVED** [review]

**Pull request merged**